core/domain/exp_domain.py

```diff
@@ -1634,7 +1634,9 @@ def validate(self, strict: bool = False) -> None:
             self._validate_state_name(state_name)
             state.validate(
                 self.param_specs,
-                allow_null_interaction=not strict)
+                allow_null_interaction=not strict,
+                tagged_skill_misconception_id_required=False,
+                empty_rule_specs_allowed=False)
```

Comment:
The default values are False, right? So do we need to pass it? (I mean in the ideal case state should not allow empty rule specs so we should not highlight it here. "ideal case" = when the question also starts doing it.)

Note: This way when we will fix issues with questions and remove the `empty_rule_specs_allowed` param we won't have to touch/change this file.

Reply:
Yes, so @vojtechjelinek suggested using `*` inside the function arguments and for that matter, I have to explicitly pass the values otherwise it will raise the error. Please let me know if it is required to be changed.
Thanks

Reply:
I think `*` only enforces that we always use keyword and not that the arg need to be always passed.